nixos/modules/services/audio/marytts.nix

```diff
@@ -0,0 +1,175 @@
+            text = '''
+              Nixpkgs | n I k s - ' p { - k @ - dZ @ s
+            ''';
+          })
+        ]
+      '';
+      description = ''
+        Paths to the user dictionary files for MaryTTS.
+      '';
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    services.marytts.settings = {
+      "socket.port" = lib.mkDefault cfg.port;
+    };
+
+    environment.systemPackages = [ pkgs.marytts ];
+
+    systemd.services.marytts = {
+      description = "MaryTTS server instance";
+      after = [ "network.target" ];
+      wantedBy = [ "multi-user.target" ];
+
+      # FIXME: MaryTTS's config loading mechanism appears to be horrendously broken
+      # and it doesn't seem to actually read config files outside of precompiled JAR files.
+      # Using system properties directly works for now, but this is really ugly.
+      script = ''
+        ${lib.getExe pkgs.marytts} ${
+          lib.concatStringsSep " " (lib.mapAttrsToList (n: v: ''-D${n}="${v}"'') cfg.settings)
+        }
+      '';
+
+      restartTriggers = cfg.voices ++ cfg.userDictionaries;
+
+      serviceConfig = {
+        DynamicUser = true;
+        User = "marytts";
+        RuntimeDirectory = "marytts";
+        StateDirectory = "marytts";
+        Restart = "on-failure";
+        RestartSec = 5;
+        TimeoutSec = 20;
+
+        # Hardening
+        ProtectClock = true;
+        ProtectKernelLogs = true;
+        ProtectControlGroups = true;
+        ProtectKernelModules = true;
+        ProtectHostname = true;
+        ProtectKernelTunables = true;
+        ProtectProc = "invisible";
+        ProtectHome = true;
+        ProcSubset = "pid";
+
+        PrivateTmp = true;
+        PrivateNetwork = false;
+        PrivateUsers = cfg.port >= 1024;
+        PrivateDevices = true;
+
+        RestrictRealtime = true;
+        RestrictNamespaces = true;
+        RestrictAddressFamilies = [
+          "AF_INET"
+          "AF_INET6"
+        ];
+
+        MemoryDenyWriteExecute = false; # Java does not like w^x :(
+        LockPersonality = true;
+        AmbientCapabilities = lib.optional (cfg.port < 1024) "CAP_NET_BIND_SERVICE";
+        CapabilityBoundingSet = "";
+        SystemCallArchitectures = "native";
+        SystemCallFilter = [
+          "@system-service"
+          "~@resources"
+          "~@privileged"
+        ];
+        UMask = "0027";
+      };
+    };
+
+    systemd.tmpfiles.settings."10-marytts" = {
+      "${maryBase}/lib"."L+".argument = "${pkgs.symlinkJoin {
+        name = "marytts-lib";
+
+        # Put user paths before default ones so that user ones have priority
+        paths = cfg.voices ++ [ "${pkgs.marytts}/lib" ];
+      }}";
+
+      "${maryBase}/user-dictionaries"."L+".argument = "${pkgs.symlinkJoin {
+        name = "marytts-user-dictionaries";
+
+        # Put user paths before default ones so that user ones have priority
+        paths = cfg.userDictionaries ++ [ "${pkgs.marytts}/user-dictionaries" ];
+      }}";
```

Comment:
```suggestion
      "${maryBase}/user-dictionaries"."L+".argument = pkgs.symlinkJoin {
        name = "marytts-user-dictionaries";

        # Put user paths before default ones so that user ones have priority
        paths = cfg.userDictionaries ++ [ "${pkgs.marytts}/user-dictionaries" ];
      };
```